src/main/kotlin/cacophony/controlflow/CFGNode.kt

```diff
@@ -290,4 +301,9 @@ sealed interface CFGNode {
         override val function: Definition.FunctionDefinition?
             get() = null
     }
+
+    data class NodeSlot<T : CFGNode>(
```

Comment:
It is possible to refactor InstructionMatcherImpl::tryMatch ifology with this kind of slot. I should have made my intentions clear that I plan to do that in another PR. 

For now at least, I think it is a good idea to have a generic NodeSlot, so that it is not necessary to create new label, new slot, and new ifs whenever we add new node slot.